ds4_driver:
  ros__parameters:
    # Only look for device with this specific hardware address.
    # If unspecified or empty string, will use the first DualShock 4 device found.
    # device_addr: 30:0E:D5:95:8D:6A

    # Whether to use hidraw or raw bluetooth mode. See docs in ds4drv for the
    # difference.
    # Options: hidraw, bluetooth
    # backend: hidraw

    # Whether to use standard ROS messages or messages defined in ds4_driver.
    # If true, will publish sensor_msgs/{Joy,BatteryState,Imu}, ds4_driver/Report,
    # and subscribe to sensor_msgs/JoyFeedbackArray.
    # If false, will publish ds4_driver/Status and subscribe to
    # ds4_driver/Feedback.
    use_standard_msgs: false

    # Rate in Hz at which a joystick that has a non-changing state will resend the
    # previously sent message. When disabled (zero), only publish Joy messages
    # when changes to axes or buttons is detected.
    # This parameter is only effective when use_standard_msgs is true.
    autorepeat_rate: 0.0

    # Amount by which the joystick has to move before it is considered to be
    # off-center.
    deadzone: 0.1

    # Maximum rate in Hz to publish Status messages.
    max_status_rate: 100.0
